assets/js/components/notifications/Notifications.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
+import DashboardLayout from '../common/DashboardLayout'
+import PaginatedTable from '../common/PaginatedTable'
+import { PAGINATED_NOTIFICATIONS, NOTIFICATION_SUBSCRIPTION } from '../../graphql/notifications'
+import { formatDatetimeAgo } from '../../util/time'
+import BlankSlate from '../common/BlankSlate'
+
+// Redux
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import  { markRead, clearAll } from '../../actions/notifications'
+
+// MUI
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import Button from '@material-ui/core/Button';
+import { withStyles } from '@material-ui/core/styles'
+
+// Icons
+import NotificationsIcon from '@material-ui/icons/Notifications';
+import ActiveIcon from '@material-ui/icons/FiberManualRecord';
+import ConsoleIcon from '../common/ConsoleIcon'
+
+const styles = theme => ({
+  paper: {
+    padding: theme.spacing.unit * 3,
+    paddingTop: theme.spacing.unit * 2,
+  },
+  header: {
+    display: 'flex',
+    justifyContent: 'space-between'
+  },
+})
+
+function mapDispatchToProps(dispatch) {
+  return bindActionCreators({ markRead, clearAll }, dispatch);
+}
+
+@withStyles(styles)
+@connect(null, mapDispatchToProps)
+class Notifications extends Component {
+  constructor(props) {
+    super(props)
+    this.markRead = this.markRead.bind(this)
+  }
+
+  markRead(notification, followUrl = true) {
+    const { markRead } = this.props
+    if (followUrl) {
+      markRead(notification.id, {active: false}, notification.url)
```

Comment:
are these params used? active: false

Reply:
oh yeah, not anymore. originally it was an updateNotification query before I changed around the data structure